### ISSUE TITLE
support partial solutions for dispatch solver

### ISSUE THREAD

**Issue by BrannonKing** [Pabloo22/job_shop_lib#3]
Feature request: allow me to give the dispatch solver a partial solution and have it finish it. Or is this already supported?

**Comment by Pabloo22** [comment]
Hello Brannon,
I'm unsure if you are referring to the `Dispatcher` class or the `DispatchingRuleSolver`.

The `DispatchingRuleSolver` already supports starting from partial solutions since both methods (`step` and `solve`) can receive an already initialized dispatcher as an argument. As a side note, this arg is optional in the `solve` method and was introduced in the latest version.

The `Dispatcher` class doesn't support being initialized from a partially completed `Schedule` yet, I could add that feature in the new version if this is what you are asking for.

**Comment by BrannonKing** [comment]
> The Dispatcher class doesn't support being initialized from a partially completed Schedule yet, I could add that feature in the new version if this is what you are asking for.

Yes, that's what I was wanting. It looks like it has an add method. Maybe the thing to do is just call the add method for my existing tasks in order.

**Comment by Pabloo22** [comment]
Yes, that's how I would implement it. You should just check that the operation is ready to be scheduled before. See `Schedule.from_job_sequences` as a reference. Internally it uses the `Dispatcher` class and does something similar to this using deques.

**Comment by dogukansert** [comment]
Can I see operation IDs or operation information on the gantt chart I created?
